packages/access-client/src/agent.js

```diff
@@ -153,25 +152,39 @@ export class Agent {
    *
    * A proof is a delegation with an audience matching agent DID
    *
-   * @param {Ucanto.Delegation} delegation
+   * @param {API.Delegation} delegation
    */
   async addProof(delegation) {
-    validate(delegation, {
-      checkAudience: this.issuer,
-      checkIsExpired: true,
-    })
-    await this.#data.addDelegation(delegation, { audience: this.meta })
+    return await this.addProofs([delegation])
+  }
+
+  /**
+   * Adds set of proofs to the agent store. Throws if the proofs are invalid.
+   * does not match the agent DID.
+   *
+   * @param {Iterable<API.Delegation>} delegations
+   */
+  async addProofs(delegations) {
```

Comment:
Can `validate` throw? If so, what happens if I pass 100 delegations, 49 are valid and then the 50th isn't, and I pass em all here?

My understanding is not only might the promise returned by this function throw (good), but when it does so, it would have already added the first 49 valid proofs to `this.#data`. i.e. there was basically partial application of `addProofs(delegations)` but without the caller knowing which part worked and which didn't.

Maybe it would be better to validate the whole batch at once in parallel before doing any `addDelegation`?

```
await Promise.all(delegations.map(proof => validate(proof, /*...*/)))
for (const proof of delegations) { this.#data.addDelegation(proof, { audience: this.meta })) }
```

Reply:
I honestly don't see value in doing validation here in first place. Even if the proofs are still valid they might expire in a minute. Furthermore `this.removeExpiredDelegations()` also goes and remove expired proofs.

I suppose we will throw errors if audience isn't the agent, but then again not sure why do we care about it.

Reply:
I have decided to remove the validation step for the following reasons

1. Time bound checks is redundant because we remove expired delegations right after we end the loop.
2. Audience check is not a good idea e.g. because we would would also want to store delegations delegated to the accounts (this agent is logged-in with). Furthermore `agent.proofs` also filters out delegations where agent is not an audience, which suggests that there is in fact assumption that audience may not be this agent.
3. If we rotate agent key we may also end up with delegations in the store where audience is not the agent, so it's best to not to have code assume that all proofs are delegated to the agent.